server/chat-plugins/lottery.js

```diff
@@ -0,0 +1,252 @@
+const lotteries = lotteriesContents ? Object.assign(Object.create(null), JSON.parse(lotteriesContents)) : Object.create(null);
+
+/**
+ * @param {string} roomid
+ * @param {number} maxWinners
+ * @param {string} name
+ * @param {string} markup
+*/
+function createLottery(roomid, maxWinners, name, markup) {
+	if (!lotteries[roomid]) {
+		lotteries[roomid] = {maxWinners, name, markup, participants: Object.create(null)};
+		writeLotteries();
+	}
+}
+function writeLotteries() {
+	FS(LOTTERY_FILE).writeUpdate(() => JSON.stringify(lotteries));
+}
+/**
+ * @param {string} roomid
+ */
+function destroyLottery(roomid) {
+	delete lotteries[roomid];
+	writeLotteries();
+}
+/**
+ * @param {string} roomid
+ * @param {User} user
+ */
+function addUserToLottery(roomid, user) {
+	const lottery = lotteries[roomid];
+	if (!lottery) return;
+	const participants = lottery.participants;
+	const userSignedup = participants[user.latestIp] || Object.values(participants).map(username => toID(username)).includes(user.userid);
+	if (!userSignedup) {
+		participants[user.latestIp] = user.name;
+		writeLotteries();
+		return true;
+	}
+	return false;
+}
+/**
+ * @param {string} roomid
+ * @param {User} user
+ */
+function removeUserFromLottery(roomid, user) {
+	const lottery = lotteries[roomid];
+	if (!lottery) return;
+	const participants = lottery.participants;
+	for (const [ip, participant] of Object.entries(participants)) {
+		if (toID(participant) === user.userid || ip === user.latestIp) {
+			delete participants[ip];
+			writeLotteries();
+			return true;
+		}
+	}
+	return false;
+}
+/**
+ * @param {string} roomid
+ */
+function getWinnersInLottery(roomid) {
+	const lottery = lotteries[roomid];
+	if (!lottery) return;
+	const winners = [];
+	const participants = Object.values(lottery.participants);
+	for (let i = 0; i < lottery.maxWinners; i++) {
+		const randomIdx = participants.length * Math.random() << 0;
+		const winner = participants[randomIdx];
+		winners.push(winner);
+		participants.splice(randomIdx, 1);
+	}
+	return winners;
+}
+
+/** @type {ChatCommands} */
+exports.commands = {
+	lottery: {
+		''(target, room) {
+			if (!room) {
+				return this.errorReply("This can only be used in rooms.");
+			}
+			const lottery = lotteries[room.id];
+			if (!lottery) {
+				return this.errorReply("This room doesn't have a lottery running.");
+			}
+			return this.parse(`/join view-lottery-${room.id}`);
+		},
+		create(target, room, user) {
+			if (!this.can('declare', null, room)) return;
+			if (!room) {
+				return this.errorReply("This can only be used in rooms.");
+			}
+			if (room.battle || !room.chatRoomData) {
+				return this.errorReply('This room does not support the creation of lotteries.');
+			}
+			if (lotteries[room.id]) {
+				return this.errorReply(`There is already a lottery running in this room.`);
+			}
+			let [maxWinners, name, ...markup] = target.split(',').map(val => val.trim());
```

Comment:
Won't this trim spaces off the html text? `Chat.splitFirst` is probably more what you're looking for here